director/director_api.go

```diff
@@ -154,3 +158,73 @@ func ConfigFilterdServers() {
 		filteredServers[sn] = permFiltered
 	}
 }
+
+// Start a goroutine to query director's Prometheus endpoint for origin/cache server I/O stats
+// and save the value to the corresponding serverAd
+func ConfigServerIOQuery(ctx context.Context, egrp *errgroup.Group) {
+	egrp.Go(func() error {
+		for {
+			select {
+			case <-ctx.Done():
+				return nil
+			case <-time.After(15 * time.Second):
+				// Requests expires before the next round starts
+				ddlCtx, cancel := context.WithDeadline(ctx, time.Now().Add(10*time.Second))
+				items := serverAds.Items()
+				wg := sync.WaitGroup{}
+				// We want to wait until all gorountines finish their work
+				wg.Add(len(items))
+				for _, item := range items {
+					if item.IsExpired() {
+						wg.Done()
+						continue
+					}
+					serverUrl := item.Key()
+					serverAd := item.Value()
+					if serverAd.FromTopology {
+						// Topology servers have no Prometheus metrics
+						wg.Done()
+						continue
+					}
+					// Run the queries in parallel, in hope that Prometheus is strong enough to handle ~100 concurrent requests
+					go func(ctx context.Context, wg *sync.WaitGroup, sUrl string, sAd *server_structs.Advertisement) {
```

Comment:
Why parallel queries?

Why not just query for *all* series in prometheus then filter out the ones not corresponding to a server we're interested in?